draft-ounsworth-pq-composite-sigs.mkd

```diff
@@ -374,11 +334,7 @@ Signature Generation Procedure:
 
 Since recursive composite public keys are disallowed in {{sec-composite-pub-keys}}, no component signature may itself be a composite; ie the signature generation routine MUST fail if one of the private keys K1, K2, .., Kn is a composite with the OID id-alg-composite.
 
-A composite signature MUST produce and include in the output a signature value for every component key in the corresponding CompositePrivateKey.
-
-EDNOTE 8: With NIST's position that they will standardize use-case-specific algorithm suites, the authors are aware of potential use-cases where a PKI entity may want to have many public keys, but only sign with a subset for each signature. At the present time, this draft does not allow for this because the algorithm for verifying "subset-signatures" in a way that is secure against algorithm stripping attacks would be very complex and prone to implementation errors (currently, the verifier can detect omitted signatures even if it does not recognize all the algorithm OIDs because the count will be wrong. In a subset-signature algorithm, additional mechanisms would be needed to specify for each component key, whether it is meant to produce a signature or not). The draft-compliant way to achieve a "subset-signature" behaviour would be for each PKI entity to have multiple public keys (and certificates) with overlapping subsets of their component keys. We welcome public opinions on whether this is sufficient, or whether this draft should specify a subset-signature algorithm.
-
-EDNOTE 9: The authors are also aware of a potential use-case of combining signature and KEM keys inside a single public key / certificate. This would give us back the "dual-usage key" property that was so appealing about RSA. At the present time, this draft does not allow for this because, again, the algorithm for verifying "subset-signatures" in a secure way would be very complex. We also welcome public opinions on this.
+A composite signature MUST produce and include in the output a signature value for every component key in the corresponding CompositePrivateKey. While it may be tempting to allow a signer to produce a signature with a subset of their keys, the complexity of securely verifying such a "subset signature" is high and out of scope for this document.
```

Comment:
I think the first sentence is self explaining. The second sentence actually doesn't give any useful information.

Reply:
Do Internet-Drafts typically justify why a decision was made, or simply state the specification?
I'm happy to take it out if this kind of thing is not usually done.

Reply:
Decision is to leave it in.